### PR TITLE
WIP: Use semantic version as identifier of Docker IQSharp images instead of build number

### PR DESCRIPTION
**Pull request by ricardo-espinoza**
This change intends to add consistency in the version numbers used for different components of the QDK.

Using the semantic version is better aligned with the names of the SDK, the NuGet packages and the VS Code extension.

### CODE REVIEW

**Comment by rmshaffer** [comment]
I just realized looking at the CI result that the CI in this repo doesn't actually use the semantic version anywhere. I think perhaps we're fine leaving this as `BUILD_BUILDNUMBER`, since that's consistent with the assemblies that get published from this repo. And the issue https://github.com/microsoft/iqsharp/issues/255 can be closed as a result of updating the official QDK pipeline to publish the images using the semantic version.

**Comment by ricardo-espinoza** [comment]
We're doing this instead in the official QDK pipeline instead.

**Comment by ricardo-espinoza** [comment]
I'm reopening the pull request, but this time changing the version identifier for the docs only.

File: build/pack-docs.ps1 [hunk 1]
```diff
@@ -30,7 +30,7 @@ if ("$Env:BUILD_RELEASETYPE" -eq "release") {
 # easily inject the right base image into the FROM line. In doing so,
 # the build context should include the build_docs.py script that we need.
 $dockerfile = @"
-FROM ${Env:DOCKER_PREFIX}iqsharp-base:${Env:BUILD_BUILDNUMBER}
+FROM ${Env:DOCKER_PREFIX}iqsharp-base:${Env:SEMVER_VERSION}
```
Comment (rmshaffer) [inline build/pack-docs.ps1:33]: I think the problem you'll run into here is that `Env:SEMVER_VERSION` is not defined in `set-env.ps1`; that's a variable that only exists in the end-to-end QDK pipeline.

Probably instead you'd want to make the image version a parameter to `pack-docs.ps1`. Then the CI build here can pass `Env:BUILD_BUILDNUMBER`, and the QDK pipeline can pass the SemVer version.

Comment (ricardo-espinoza) [reply build/pack-docs.ps1:33]: Makes sense. I'll start that approach as a separate PR.

**Pull request closed without merging**